### PR TITLE
`Input` form atom

### PR DESCRIPTION
**Pull request by zachstence**
image

### CODE REVIEW

File: clients/web/package.json [hunk 1]
```diff
@@ -29,6 +30,7 @@
     "histoire": "^0.11.0",
     "postcss": "^8.4.14",
     "postcss-load-config": "^4.0.1",
+    "shortid": "^2.2.16",
```
Comment (0c370t) [inline clients/web/package.json:33]: Can we use nanoid to match what's used elsewhere?

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    export let placeholder: string | undefined = undefined;
+    export let disabled: boolean = false;
+    export let state: FormControlState = "none";
+    export let error: string | undefined = undefined;
+
+    const labelId = `input_${shortid.generate()}`;
+    const errorId = `input-error_${shortid.generate()}`;
+</script>
+
+<div class="size-{size} state-{state}">
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/Input/Input.svelte:22]: If size and state are missing; do we want to continue attaching these classes?

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    label {
+        @apply block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300;
+    }
+
+    .input-container {
+        @apply flex items-stretch
+            rounded-lg border overflow-hidden
+            focus-within:ring-1 focus-within:ring-primary focus-within:border-primary
+            bg-gray-50 border-gray-300 text-gray-900 placeholder-gray-600
+            dark:bg-gray-700 dark:border-gray-600 dark:text-white dark:placeholder-gray-400;
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/Input/Input.svelte:57]: We are dark mode by default; without support for light mode, so we should swap this to remove the light mode values

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte [hunk 4]
```diff
@@ -0,0 +1,129 @@
+<script lang="ts" context="module">
+    export type InputSize = "sm" | "md" | "lg";
```
Comment (0c370t) [inline clients/web/src/lib/components/atoms/form/Input/Input.svelte:2]: Can this be consolidated into the form types?

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte (hunk 2 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte:22]: Size and state have default values of `md` and `none` respectively, so they won't be missing, and we do want to apply the classes

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte (hunk 3 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte:57]: Why not support light mode? Its built in to all of the Flowbite components automatically

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte (hunk 4 above)
Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte]: https://github.com/SprocketBot/sprocket/pull/333/commits/82769efee0558dcc49223eb64f156bf0ecd5adb7

File: clients/web/src/lib/components/atoms/form/Input/Input.svelte (hunk 3 above)
Comment (0c370t) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte:56]: The effort of supporting it both from a design & implementation perspective doesn't make sense (imo) compared to the potential for issues (i.e. a component written in one that doesn't have the responsive styles).

I would guess that the _overwhelming_ majority of our users are likely to use dark mode anyways.

Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte:56]: Fair, will remove

Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte:56]: I'd love to support light mode in the future though. Some people (particularly those with disabilities) strongly prefer light mode

Comment (zachstence) [reply clients/web/src/lib/components/atoms/form/Input/Input.svelte]: Fixed https://github.com/SprocketBot/sprocket/pull/333/commits/68996377c29f80fe013b265e8ac4d73998792f1d

**Review by 0c370t — APPROVED** [review]

**Pull request merged**